ai/recommender/surprise_recommenders.py

```diff
@@ -1,11 +1,17 @@
 # Recommender system for Penn AI.
 import pandas as pd
+from pandas.util import hash_pandas_object
```

Comment:
This issue is related to `CoClusteringRecommender` from surprise package. I got a lot of stdout about "on user XXX" and "on term XXX" as below. Could you please turn it off?

```
surprise_recommenders: DEBUG: self.algo_name: CoClustering
surprise_recommenders: DEBUG: ml_type: classifier
surprise_recommenders: DEBUG: metric: accuracy
base: INFO: self.random_state: 42
base: INFO: load_serialized_rec='if_exists'
base: WARNING: Not loading serialized recommender, file does not exist: {self.serialized_rec_path}
base: INFO: updating hash_2_param...
base: INFO: storing parameter hash...
surprise_recommenders: DEBUG: updating CoClustering model
surprise_recommenders: DEBUG: shuffling results_data
surprise_recommenders: INFO: append and drop dupes
surprise_recommenders: INFO: load_from_df
surprise_recommenders: DEBUG: self.trainset # of ML-P combos: 381
surprise_recommenders: DEBUG: self.trainset # of datasets: 164
surprise_recommenders: DEBUG: fitting self.algo...
calling AlgoBase.fit
called AlgoBase.fit
on user 0
trainset.ur[0]: [(0, 0.979), (240, 0.118), (228, 0.948), (302, 0.652), (296, 0.974), (112, 0.983), (195, 0.974), (184, 0.968), (118, 0.745), (185, 0.978), (21, 0.687), (74, 0.974), (42, 0.977), (208, 0.17), (339, 0.978), (102, 0.974), (114, 0.2), (113, 0.8909999999999999), (255, 0.977), (231, 0.2), (190, 0.978), (332, 0.974), (75, 0.89), (37, 0.979), (23, 0.974), (283, 0.977), (218, 0.6940000000000001), (198, 0.963), (264, 0.7190000000000001), (209, 0.953), (31, 0.112), (10, 0.966), (249, 0.981), (292, 0.968), (189, 0.982), (289, 0.979), (342, 0.978), (285, 0.962), (89, 0.244), (158, 0.328), (127, 0.916), (369, 0.978), (145, 0.151), (57, 0.978)]
value: [0.979, 0.118, 0.948, 0.652, 0.974, 0.983, 0.974, 0.968, 0.745, 0.978, 0.687, 0.974, 0.977, 0.17, 0.978, 0.974, 0.2, 0.8909999999999999, 0.977, 0.2, 0.978, 0.974, 0.89, 0.979, 0.974, 0.977, 0.6940000000000001, 0.963, 0.7190000000000001, 0.953, 0.112, 0.966, 0.981, 0.968, 0.982, 0.979, 0.978, 0.962, 0.244, 0.328, 0.916, 0.978, 0.151, 0.978]
on user 1
trainset.ur[1]: [(1, 0.867), (35, 0.8590000000000001), (217, 0.8590000000000001), (255, 0.911), (264, 0.8590000000000001), (184, 0.885), (175, 0.8859999999999999), (354, 0.915), (94, 0.898), (47, 0.858), (186, 0.943), (2, 0.91), (92, 0.893), (74, 0.8859999999999999), (128, 0.917), (327, 0.889), (299, 0.8590000000000001), (216, 0.899), (114, 0.8590000000000001), (43, 0.845), (155, 0.957), (323, 0.8590000000000001), (30, 0.898), (77, 0.852), (313, 0.8590000000000001), (158, 0.911), (333, 0.879), (165, 0.6709999999999999)]
value: [0.867, 0.8590000000000001, 0.8590000000000001, 0.911, 0.8590000000000001, 0.885, 0.8859999999999999, 0.915, 0.898, 0.858, 0.943, 0.91, 0.893, 0.8859999999999999, 0.917, 0.889, 0.8590000000000001, 0.899, 0.8590000000000001, 0.845, 0.957, 0.8590000000000001, 0.898, 0.852, 0.8590000000000001, 0.911, 0.879, 0.6709999999999999]
on user 2
trainset.ur[2]: [(2, 0.989), (30, 0.993), (134, 0.995), (123, 0.984), (0, 0.994), (275, 0.993), (307, 0.962), (335, 0.604), (279, 0.991), (235, 0.974), (118, 0.696), (291, 0.522), (363, 0.993), (165, 0.974), (70, 0.992), (343, 0.993), (221, 0.991), (7, 0.982), (29, 0.522), (232, 0.942), (75, 0.993), (243, 0.996), (167, 0.938), (77, 0.993), (64, 0.993), (49, 0.899), (137, 0.993), (99, 0.993), (219, 0.522), (194, 0.993), (48, 0.993), (283, 0.994), (297, 0.993), (201, 0.993), (98, 0.99), (124, 0.5710000000000001), (132, 0.696), (133, 0.95), (47, 0.918), (145, 0.975), (89, 0.899), (357, 0.989)]
value: [0.989, 0.993, 0.995, 0.984, 0.994, 0.993, 0.962, 0.604, 0.991, 0.974, 0.696, 0.522, 0.993, 0.974, 0.992, 0.993, 0.991, 0.982, 0.522, 0.942, 0.993, 0.996, 0.938, 0.993, 0.993, 0.899, 0.993, 0.993, 0.522, 0.993, 0.993, 0.994, 0.993, 0.993, 0.99, 0.5710000000000001, 0.696, 0.95, 0.918, 0.975, 0.899, 0.989]
...
on item 375 : [0.58, 0.649, 0.657, 0.951, 0.867, 0.951, 0.85, 0.952, 0.86, 0.491, 0.795, 0.96, 0.557]
on item 376 : [0.541, 0.535, 0.625, 0.802, 0.561, 0.779, 0.794, 0.371, 0.657, 0.654, 0.562]
on item 377 : [0.769, 0.83, 0.7390000000000001, 0.84, 0.883, 0.992, 0.759, 0.5]
on item 378 : [0.953, 0.743, 0.97, 0.995, 0.7070000000000001, 0.7240000000000001, 0.77, 0.512, 0.961, 0.094, 0.856, 1.0, 1.0]
on item 379 : [0.56, 0.654, 0.875, 0.851]
on item 380 : [0.945, 0.961, 0.992, 0.638, 0.877, 0.643, 0.544, 0.98, 0.8240000000000001, 0.52]
```

Reply:
A lot of logs... check [this](https://travis-ci.org/github/EpistasisLab/pennai/jobs/723883564#L21871)

Reply:
Please clean up those `print`s in [your surprise repo](https://github.com/lacava/Surprise/blob/1.0.8.2/surprise/prediction_algorithms/co_clustering.pyx#L104-L109) for fixing this issue.

Reply:
addressed this in https://github.com/lacava/Surprise/releases/tag/1.0.8.3